python/hyperon/base.py

```diff
@@ -19,84 +23,128 @@ def query(self, query_atom):
     #     None
 
     def add(self, atom):
+        """
+        Adds an Atom to the atom space. Must be implemented in derived classes.
+        """
         raise RuntimeError("Space::add() is not implemented")
 
     def remove(self, atom):
+        """
+        Removes an Atom from the atom space. Must be implemented in derived classes.
+        """
         raise RuntimeError("Space::remove() is not implemented")
 
     def replace(self, atom, replacement):
+        """
+        Replaces an Atom from the atom space. Must be implemented in derived classes.
+        """
         raise RuntimeError("Space::replace() is not implemented")
 
     def atom_count(self):
+        """
+        Counts the number of atoms in the atom space. Optional for derived classes.
+        """
         None
 
     def atoms_iter(self):
+        """
+        Returns an iterator over atoms in the Space. Optional for derived classes.
+        """
         None
 
-"""
-A wrapper over the native GroundingSpace implementation, that can be subclassed and extended within Python
-"""
 class GroundingSpace(AbstractSpace):
-
+    """
+    A wrapper over the native GroundingSpace implementation, which can be subclassed
+    and extended within Python
+    """
     def __init__(self, unwrap=True):
+        """Initialize GroundingSpace and its underlying native implementation."""
         super().__init__()
         # self.cspace = hp.space_new_grounding()
         self.gspace = GroundingSpaceRef()
 
     def query(self, query_atom):
+        """
+        Delegates the query to the underlying native GroundingSpace
+        and returns the result BindingsSet
+        """
         return self.gspace.query(query_atom)
 
     # TODO (INTERNAL): Currently unimplemented.
     # def subst(self, pattern, templ):
 
     def add(self, atom):
+        """
+        Adds an Atom to the atom space.
+        """
         self.gspace.add_atom(atom)
 
     def remove(self, atom):
+        """
+        Removes an Atom from the atom space.
+        """
         return self.gspace.remove_atom(atom)
 
     def replace(self, from_atom, to_atom):
+        """
+        Replaces an Atom in the atom space.
+        """
         return self.gspace.replace_atom(from_atom, to_atom)
 
     def atom_count(self):
+        """
+        Counts the number of Atoms in the atom space.
+        """
         return self.gspace.atom_count()
 
     def atoms_iter(self):
+        """
+        Returns an iterator over atoms in the atom space.
+        """
         return iter(self.gspace.get_atoms())
 
-"""
-Private glue for Hyperonpy implementation
-"""
 def _priv_call_query_on_python_space(space, query_catom):
+    """
+    Private glue for Hyperonpy implementation.
+    Translates a native 'catom' into an Atom object, and then delegates the query
```

Comment:
Do we really want to explain the behavior of our private glue functions?

Reply:
We may not necessarily, but if it is here, we can keep it.